File: Data Structures & Algorithms/count-number-of-islands/submission-0.cpp
class Solution {
private:
    int dfs(int row, int col, int width, int height, const vector<vector<char>>& grid, vector<vector<bool>>& visited) {
        if (row < 0 || row >= height || col < 0 || col >= width) {
            return 0;
        }
        if (visited[row][col] || grid[row][col] == '0') {
            return 0;
        }
        visited[row][col] = true;
        
        vector<int> dRow = {0, 0, -1, 1};
        vector<int> dCol = {-1, 1, 0, 0};
        for (int d = 0; d < 4; ++d) {
            dfs(row + dRow[d], col + dCol[d], width, height, grid, visited);
        }

        return 1;
    }

public:
    int numIslands(vector<vector<char>>& grid) {
        int height = grid.size();
        int width = grid[0].size();
        int num_islands = 0;
        vector<vector<bool>> visited(height, vector<bool>(width, false));

        for (int row = 0; row < height; ++row) {
            for (int col = 0, width = grid[row].size(); col < width; ++col) {
                num_islands += dfs(row, col, width, height, grid, visited);
            }
        }

        return num_islands;
    }
};

File: Data Structures & Algorithms/valid-sudoku/submission-0.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        height = 9
        width = 9
        boxwidth = 3
        def validRows():
            for row in range(height):
                alreadyExists = [False] * 10
                for col in range(width):
                    if board[row][col] == ".":
                        continue
                    num = int(board[row][col])
                    if alreadyExists[num]:
                        return False
                    alreadyExists[num] = True
            return True

        def validCols():
            for col in range(width):
                alreadyExists = [False] * 10
                for row in range(height):
                    if board[row][col] == ".":
                        continue
                    num = int(board[row][col])
                    if alreadyExists[num]:
                        return False
                    alreadyExists[num] = True
            return True
        
        def validSubBoxes():
            for box in range(9):
                cornerRow = (box // 3) * 3
                cornerCol = (box % 3) * 3
                alreadyExists = [False] * 10
                for row in range(cornerRow, cornerRow + 3):
                    for col in range(cornerCol, cornerCol + 3):
                        if board[row][col] == ".":
                            continue
                        num = int(board[row][col])
                        if alreadyExists[num]:
                            print(box, num, alreadyExists)
                            return False
                        alreadyExists[num] = True
            return True
            
        """
        box 0, 1, 2, 3, 4, 5, 6, 7, 8
        col 0, 3, 6, 0, 3, 6, 0, 3, 6 = (box % 3) * 3
        row 0, 0, 0, 3, 3, 3, 6, 6, 6 = (box // 3) * 3
        """
        
        if not validRows():
            print("ROW")
            return False
        if not validCols():
            print("COL")
            return False
        if not validSubBoxes():
            print("BOX")
            return False
        return True